C Program to print  m*n matrix in spiral order.

narayan2111 opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given a matrix of m x n elements (m rows, n columns), return all elements of the matrix in spiral order.

Sample case 1:

	Input:

	Enter Rows:2
	Enter Column: 2
	1
	2
	3
	4

	Output:

	Entered Matrix:
	1 2 
	3 4

	Spiral Order of the Matrix: 1 2 4 3

Sample Case 2:

	Input:

	Enter Rows:2
	Enter Column: 2
	0
	0
	0
	0

	Output:

	Null Matrix

Time Complexity: O(ROW*COLUMN)
Space Complexity:O(1).

### Checklist

- [Y ] I've read the contribution guidelines.
- [ Y] I've checked the issue list before deciding what to submit.
- [ Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#792 
![ProgramOutput](https://user-images.githubusercontent.com/66173390/91830168-0bf76680-ec60-11ea-9627-961e5c2cf17d.jpg)

narayan2111 committed:
Merge pull request #1 from TesseractCoding/master

Updating the forked repo

narayan2111 committed:
Added Spiral Matrix

narayan2111 committed:
Updated README.md file

narayan2111 commented:
Kindly Review this PR. @Abhijit2505 @plazzy99 @iamrajiv @Kajol-Kumari

narayan2111 committed:
Merge branch 'master' into spiralmatrix_issue792

vatsalkesarwani12 reviewed (approved):
Code works fine for all test case 👍 
Able to merge

atarax665 commented on C/README.md:
```diff
@@ -1,3 +1,75 @@
+
```
You only need to add your program's hyperlink in the README.md file. Check out other PRs for reference.

atarax665 reviewed (changes requested):
Code looks good. Do the suggested changes in readme ASAP so that I can merge this PR.

narayan2111 commented:
> Code looks good. Do the suggested changes in readme ASAP so that I can merge this PR.

How will I do that  sir? 
I mean it works fine and i also updated the readme file and also added link of the code.

narayan2111 commented:
Kindly review this PR, it's pending from 4 days @atarax665

Abhijit2505 reviewed (approved):
Code looks good, just make the ReadMe changes asked before.
